appstore-metadata-service/src/main/java/com/lgi/appstore/metadata/api/maintainer/PersistentAppsService.java

```diff
@@ -412,6 +413,10 @@ public boolean updateApplication(String maintainerCode, String appId, String ver
                             .map(integerRecord1 -> integerRecord1.get(MAINTAINER.ID))
                             .orElseThrow(() -> new MaintainerNotFoundException(maintainerCode));
 
+                    final String newVersion = isBlank(applicationForUpdate.getHeader().getNewVersion())
+                            ? version
+                            : applicationForUpdate.getHeader().getNewVersion();
```

Comment:
this logic could be placed in applicationForUpdate.getApplicationVersion() or to header class

Reply:
Unfortunately `ApplicationForUpdate` class and header class are generated by swagger file when building the project, so I can create a separate method in that class.